File: app/src/main/java/com/tkamat/android/cyou/TopicRecyclerActivity.java
package com.tkamat.android.cyou;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;

/**
 * @author Tushaar Kamat
 * @version 9/8/17
 */

public class TopicRecyclerActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_fragment);

        FragmentManager fm = getSupportFragmentManager();
        Fragment fragment = fm.findFragmentById(R.id.fragment_container);

        if (fragment == null) {
            fragment = TopicRecyclerFragment.newInstance();
            fm.beginTransaction().add(R.id.fragment_container, fragment).commit();
        }
    }

}

File: app/src/main/java/com/tkamat/android/youwatch/TopicRecyclerActivity.kt
package com.tkamat.android.youwatch

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment

/**
 * @author Tushaar Kamat
 * @version 9/8/17
 */

class TopicRecyclerActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_fragment)

        val fm = supportFragmentManager
        var fragment: Fragment? = fm.findFragmentById(R.id.fragment_container)

        fragment ?: run {
            fragment = TopicRecyclerFragment.newInstance()
            fm.beginTransaction().add(R.id.fragment_container, fragment as TopicRecyclerFragment).commit()
        }
    }

}
